### PR TITLE
chore: Store EXTENSION_INFO as Set[String] instead of newline-delimited String

### PR DESCRIPTION
**Pull request by andygrove**
## Which issue does this PR close?

N/A

## Rationale for this change

I think it would be cleaner to store extension info as `Set[String]` rather than using a newline-delimited string, because:

- It removes the need to use `split("\n")` and `mkString("\n")` when manipulating the info
- It ensures that we do not have duplicate entries

## What changes are included in this PR?

- Some refactoring inside `CometSparkSessionExtensions`

## How are these changes tested?

Existing tests

### CODE REVIEW

**Comment by andygrove** [comment]
@parthchandra Does this seem reasonable?

**Comment by andygrove** [comment]
Thanks for the reviews @parthchandra and @advancedxy

@viirya I need a committer approval to be able to merge. Could you review?

**Review by viirya — APPROVED** [review]
Looks good to me. Thanks @andygrove

**Pull request merged**
